src/qutip_qip/device/cavityqed.py

```diff
@@ -147,7 +147,14 @@ def eliminate_auxillary_modes(self, U):
             [basis(self.num_levels, 0)]
             + [identity(2) for n in range(self.num_qubits)]
         )
-        return psi_proj.dag() * U * psi_proj
+        result = psi_proj.dag() * U * psi_proj
+        # In qutip 5 multiplication of matrices
+        # with dims [[1, 2], [2, 2]] and [[2, 2], [1, 2]]
+        # will give a result of
+        # dims [[1, 2], [1, 2]] instead of [[2], [2]].
+        if result.dims[0][0] == 1:
+            result = result.ptrace(list(range(len(self.dims)))[1:])
```

Comment:
This looks like it will bite other users. Is there some better way to support it? Maybe `psi_proj` dims need to be fixed before the multiplication somehow?

Reply:
Since we'd added looking at this to the TODO list, let's leave it for now.

Reply:
@hodgestar, @Ericgig, I was collecting some common error messages that I saw while updating to v5. And I notice this again. This issue still remains. It is still somewhere in a TODO list?